packages/x-components/src/utils/object.ts

```diff
@@ -138,3 +138,29 @@ export function getNewAndUpdatedKeys<ObjectType extends Dictionary>(
 
   return Object.keys(newValue).filter(key => !(key in oldValue) || newValue[key] !== oldValue[key]);
 }
+
+/**
+ * Ensures that the given condition is met in all the entries of the object.
+ *
+ * @param object - The object to check if every item meets the given condition.
+ * @param condition - The condition to check in each one of the entries of the object.
+ *
+ * @returns True when all the entries pass the condition. False otherwise.
+ * @public
+ */
+export function every<ObjectType extends Dictionary>(
```

Comment:
Initially I had manually stubbed the navigator `sendBeacon` method, and then used this `every` method to perform checks in the tests. Now this is not needed anymore, but I think this function can help us in the future.

Reply:
Hey bro! You're human! Welcome to the human realm! :trollface:

In a more serious note, we are fine with the addition of this utility. However, it currently has a bug. The `paramsMatch` flag "accumulator" is not working as intended, as it is being reset each iteration. We should not invert its value and we should use an `&&` operator instead of the `||` one, so the previous value is always taken into account. For example, in case in a previous iteration `paramsMatch` resulted in a `false` value, we shouldn't even care about the `condition` of the current/next ones, as at least one object key/entry has already failed the condition.

You can clearly see this behaviour happening in the second test you made, if you invert the order in the object, { a: '1', b: 2 }, you will see that the test fails, where it shouldn't, as we are only checking, in the end, that the last entry/key fulfils the condition.

As a side note, we could even optimise this method so it breaks the reduce execution whenever the condition is not met (probably not worth it for the majority of use cases, tho).